### PR TITLE
feat(tanstack-start): Add `getAuth` server function

### PR DESCRIPTION
**Pull request by octoper**
## Description

This PR introduces `getAuth` server function that can be used inside the TanStack start `createServerFn`.

Example:

```tsx
import { getAuth } from '@clerk/tanstack-start/server'
import { createServerFn } from '@tanstack/start'


export const fetchCurrentUserPosts = createServerFn('GET', async (_payload, ctx) => {
  const { userId } = await getAuth(ctx)

  if (!userId) {
    ...
  }

  ...
})
```

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Review by wobsoriano — APPROVED** [review]
Looking good 🚀

File: packages/tanstack-start/src/server/getAuth.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { loadOptions } from './loadOptions';
+import type { LoaderOptions } from './types';
+
+export type GetAuthReturn = Promise<AuthObject>;
+
+type GetAuthOptions = Pick<LoaderOptions, 'secretKey'>;
+
+export async function getAuth(ctx: FetchFnCtx, opts?: GetAuthOptions): GetAuthReturn {
+  if (!ctx || (ctx && !ctx.request)) {
+    throw new Error(noFetchFnCtxPassedInGetAuth);
```
Comment (panteliselef) [inline packages/tanstack-start/src/server/getAuth.ts:16]: ❓Should we use `errorThrower` here?

File: packages/tanstack-start/src/server/getAuth.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import type { AuthObject } from '@clerk/backend/internal';
+import { stripPrivateDataFromObject } from '@clerk/backend/internal';
+import type { FetchFnCtx } from '@tanstack/start';
+
+import { noFetchFnCtxPassedInGetAuth } from '../utils/errors';
+import { authenticateRequest } from './authenticateRequest';
+import { loadOptions } from './loadOptions';
+import type { LoaderOptions } from './types';
+
+export type GetAuthReturn = Promise<AuthObject>;
```
Comment (panteliselef) [inline packages/tanstack-start/src/server/getAuth.ts:10]: ❓Do we need to export this type ?

File: packages/tanstack-start/package.json [hunk 3]
```diff
@@ -58,18 +58,18 @@
     "@clerk/clerk-react": "5.3.1",
     "@clerk/shared": "2.4.3",
     "@clerk/types": "4.11.0",
-    "@vinxi/server-functions": "^0.3.2",
```
Comment (panteliselef) [inline packages/tanstack-start/package.json:61]: 💯

**Review by panteliselef — APPROVED** [review]

File: packages/tanstack-start/src/server/getAuth.ts (hunk 2 above)
Comment (octoper) [reply packages/tanstack-start/src/server/getAuth.ts:10]: Good catch, removing!

**Review by nikospapcom — APPROVED** [review]

**Pull request merged**